### PR TITLE
new read-in function

### PR DESCRIPTION
**Pull request by ktoddbrown**
Previously when a function was read-in or 'processes' with a dataset-key we tried to parse the data into 5 data frames, some of these data frames were wide and others were long. After several discussions with people over the past few months I've simplified this process to two data frames, one wide and one long. This simplifies the function significantly but transfers more work outside the function; mainly it's now on the user to break out the sub-data frames.

Given the eccentricities of each data set I think this works better then trying to cetch general cases inside the core utility functions here.

Thoughts?

### CODE REVIEW

File: R/readKeyedData.R [hunk 1]
```diff
@@ -0,0 +1,203 @@
...
+    }
+  }else if(readingEXCELworkbook){
+    numIterate <- length(readxl::excel_sheets(filename))
+    if(any(readxl::excel_sheets(filename) %in% verticalTable)){
+      verticalTable <- which(readxl::excel_sheets(filename) %in% verticalTable)
+    }else{
+      verticalTable <- -1
+    }
+  }else{
+    stop('Invalide file format, must be csv files or an excel workbook')
```
Comment (bpbond) [inline R/readKeyedData.R]: "Invalid"

File: R/readKeyedData.R [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+  assertthat::assert_that(any(is.na(key.df$softType) & !is.na(key.df$flagID)), 
+             msg='A long data type can not be a flaged id')
+  assertthat::assert_that(any(!is.na(key.df$softType) & is.na(key.df$flagID)),
+             msg='There must be a flagged id for the wide data types')
+  
+  
+  wide.df <- NULL
+  long.df <- NULL
+  
+  for(tableIndex in 1:numIterate){
```
Comment (bpbond) [inline R/readKeyedData.R]: `seq_len(numIterate)` safer

File: R/readKeyedData.R [hunk 3]
```diff
@@ -0,0 +1,203 @@
...
+    }
+    
+    ### pull the wide data ###
+    wideKey.df <- key.df %>%
+      filter(is.na(softType), ##define wide tables by absence of soft Type
+             header %in% names(data.df)) ##only pull the headers we care about
+    
+    wideTemp <- data.df %>% 
+      select_(quote(wideKey.df$header)) %>%
+      select_if(~any(!is.na(.))) #drop empty columns
```
Comment (bpbond) [inline R/readKeyedData.R:129]: Whoah funky

**Review by bpbond — APPROVED** [review]
Overall this seems like a much cleaner and clearer way to do things.

File: R/readKeyedData.R (hunk 3 above)
Comment (ktoddbrown) [reply R/readKeyedData.R:129]: I know right? I was delighted to stumble across this little gem in my frantic google code searches.

File: R/readKeyedData.R (hunk 1 above)
Comment (ktoddbrown) [reply R/readKeyedData.R]: Thanks! Fixed on commit bd611d822cbbb0b9d128daa0840003bd60cc006e

File: R/readKeyedData.R (hunk 2 above)
Comment (ktoddbrown) [reply R/readKeyedData.R]: Thanks! See commit ef6833ec0204451a81d8a2aa64382b4e9c136954

**Pull request merged**